File: app/src/main/java/com/rafaeltmbr/stopwatch/domain/services/TimerService.kt
package com.rafaeltmbr.stopwatch.domain.services

import kotlinx.coroutines.flow.StateFlow

data class TimerState(
    val isRunning: Boolean,
    val milliseconds: Long
)

interface TimerService {
    val state: StateFlow<TimerState>

    fun start()
    fun pause()
    fun reset()
}

File: ios/Stopwatch/core/services/TimerService.swift
import Foundation

struct TimerState {
    let isRunning: Bool
    let milliseconds: Int
    
    init(isRunning: Bool = false, milliseconds: Int = 0) {
        self.isRunning = isRunning
        self.milliseconds = milliseconds
    }
}

protocol TimerService {
    associatedtype Events: EventSubscriber where Events.Event == TimerState
    
    var state: TimerState { get }
    var events: Events { get }
    
    func set(_ state: TimerState)
    func start()
    func pause()
    func reset()
}
